### ISSUE TITLE
unusual performance on Rim when the sensitive attribute is Gender

### ISSUE THREAD

**Issue by string-ellipses** [Harvard-Ophthalmology-AI-Lab/FairSeg#10]
![image](https://github.com/user-attachments/assets/e9d51838-3e3d-4d23-a0f6-835ec797913f)

I have reproduced the experiment on HarvardFairSeg Dataset, with the split following the files _SAMed/lists/FairSeg_final/train.txt&test.txt_

The model used is SAMed (ViT-B) with a learning rate of 0.005, momentum of 0.9, weight decay of 0.1, a batch size of 42, and trained for 70 epochs (as the implementation setting in the paper) on A800.（for both SAMed and SAMed+FEBS) 

However, the results seem unusual. As shown in the image, for the Cup, SAMed+FEBS outperforms SAMed across all metrics; for the Rim,  SAMed+FEBS performed worse than SAMed across all metrics?

**Comment by tianyu0207** [comment]
> ![image](https://private-user-images.githubusercontent.com/126571562/386486465-e9d51838-3e3d-4d23-a0f6-835ec797913f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM4NTYzMTMsIm5iZiI6MTczMzg1NjAxMywicGF0aCI6Ii8xMjY1NzE1NjIvMzg2NDg2NDY1LWU5ZDUxODM4LTNlM2QtNGQyMy1hMGY2LTgzNWVjNzk3OTEzZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxMFQxODQwMTNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hOTgyZWQ4MWU3ZmFkM2Q1NTg2YTQyZGUwMmU5NTY5MGU1NzQ0NGE4OWE1ZWYyZjUyNzU5YjgwYjBlZTRlNDg4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.4VhqJggSKU0j-vaXUrGPU8PE8cAj5USeGirWTDfq-Eo)
> 
> I have reproduced the experiment on HarvardFairSeg Dataset, with the split following the files _SAMed/lists/FairSeg_final/train.txt&test.txt_
> 
> The model used is SAMed (ViT-B) with a learning rate of 0.005, momentum of 0.9, weight decay of 0.1, a batch size of 42, and trained for 70 epochs (as the implementation setting in the paper) on A800.（for both SAMed and SAMed+FEBS)
> 
> However, the results seem unusual. As shown in the image, for the Cup, SAMed+FEBS outperforms SAMed across all metrics; for the Rim, SAMed+FEBS performed worse than SAMed across all metrics?

Hi Thanks for your interests. Could you try to tune the hyper-parameter to see if you could improve the results. Given that there are differences with the environment/ GPU or CUDA versions/ Pytorch, Python version/ and some hidden differences. There might be some of the differences for the results compared to the original paper.
